pkg/workflows/wasm/v2/runner_test_hooks.go

```diff
@@ -0,0 +1,46 @@
+//go:build !wasip1
+
+package wasm
+
+import (
+	"sync"
+	"testing"
+	"unsafe"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows/sdk/v2"
+	sdkpb "github.com/smartcontractkit/chainlink-common/pkg/workflows/sdk/v2/pb"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows/sdk/v2/testutils"
+)
+
+// initRunnerAndRuntimeForTest is NOT thread safe and expects a single test to be running when testing the runner.
+func initRunnerAndRuntimeForTest(t testing.TB, execId string) {
+	lock.Lock()
+	defer lock.Unlock()
+	registry = testutils.GetRegistry(t)
+	outstandingCalls = map[string]sdk.Promise[*sdkpb.CapabilityResponse]{}
+	testTb = t
+	executionId = execId
+	callCapabilityErr = false
+	t.Cleanup(func() {
+		lock.Lock()
+		defer lock.Unlock()
+		testTb = nil
+		executionId = ""
+		outstandingCalls = nil
+	})
+}
+
+var lock sync.Mutex
```

Comment:
Why are these all globals?

Reply:
The goal of the test hooks are to allow testing to be as close as possible to the WASI calls.  To achieve this, I wrote a test method for each WASI call. The WASI calls are all exposed like C calls so they are global. Since they are global, I needed the global test locks.